RELEASE.md

```diff
@@ -0,0 +1,47 @@
+# Release Process
+
+The Controller-Runtime Project is released on an as-needed basis. The process is as follows:
```

Comment:
"The controller-runtime Project" or "The Kubernetes controller-runtime Project" may be better?